ndcube/ndcube.py

```diff
@@ -128,6 +128,38 @@ def crop_by_values(self, lower_corner, upper_corner, units=None, wcs=None):
         """
 
 
+class NDCubeLinkedDescriptor:
+    """
+    A descriptor which gives the property a reference to the cube to which it is attached.
+    """
+    def __init__(self, default_type):
+        self._default_type = default_type
+        self._property_name = None
+
+    def __set_name__(self, owner, name):
+        self._property_name = name
+        self._attribute_name = f"_{name}"
```

Comment:
Might be helpful to add a comment here explaining different between `_property_name` and `_attribute_name`.